cvat-canvas3d/src/typescript/canvas3dView.ts

```diff
@@ -73,14 +201,90 @@ export class Canvas3dViewImpl implements Canvas3dView, Listener {
-        this.scene.add(points);
+        this.views.perspective.scene.add(points);
+        this.views.top.scene.add(points.clone());
+        this.views.side.scene.add(points.clone());
+        this.views.front.scene.add(points.clone());
+    }
+
+    private static resizeRendererToDisplaySize(view: RenderView): void {
+        const canvas = view.renderer.domElement;
+        const width = canvas.parentElement.clientWidth;
+        const height = canvas.parentElement.clientHeight;
+        const needResize = canvas.clientWidth !== width || canvas.clientHeight !== height;
+        if (needResize) {
+            // eslint-disable-next-line no-param-reassign
+            view.camera.aspect = width / height;
+            view.camera.updateProjectionMatrix();
+            view.renderer.setSize(width, height);
+        }
     }
 
     public render(): void {
-        this.renderer.render(this.scene, this.camera);
+        this.views.perspective.controls.update(this.clock.getDelta());
+        this.views.side.controls.update();
+        this.views.top.controls.update();
+        this.views.front.controls.update();
+        Canvas3dViewImpl.resizeRendererToDisplaySize(this.views.perspective);
+        Canvas3dViewImpl.resizeRendererToDisplaySize(this.views.top);
+        Canvas3dViewImpl.resizeRendererToDisplaySize(this.views.side);
+        Canvas3dViewImpl.resizeRendererToDisplaySize(this.views.front);
+        this.views.perspective.renderer.render(this.views.perspective.scene, this.views.perspective.camera);
+        this.views.top.renderer.render(this.views.top.scene, this.views.top.camera);
+        this.views.side.renderer.render(this.views.side.scene, this.views.side.camera);
+        this.views.front.renderer.render(this.views.front.scene, this.views.front.camera);
+    }
+
+    public keyControls(key: any): void {
+        if (key.altKey === true) {
+            switch (key.code) {
+                case CAMERA_ACTION.ZOOM_IN:
+                    this.views.perspective.controls.dolly(5, true);
+                    break;
+                case CAMERA_ACTION.ZOOM_OUT:
+                    this.views.perspective.controls.dolly(-5, true);
+                    break;
+                case CAMERA_ACTION.MOVE_LEFT:
+                    this.views.perspective.controls.truck(-0.01 * this.speed, 0, true);
+                    break;
+                case CAMERA_ACTION.MOVE_RIGHT:
+                    this.views.perspective.controls.truck(0.01 * this.speed, 0, true);
+                    break;
+                case CAMERA_ACTION.MOVE_DOWN:
+                    this.views.perspective.controls.truck(0, -0.01 * this.speed, true);
+                    break;
+                case CAMERA_ACTION.MOVE_UP:
+                    this.views.perspective.controls.truck(0, 0.01 * this.speed, true);
+                    break;
+                default:
+                    break;
+            }
+        } else {
```

Comment:
Maybe we can move it up to make it possible to work keyboard arrows with Alt as well, to make moving easer for user, and handle Alt+ combinations with single if after

Reply:
Resolved